src/shared/mkfs-util.c

```diff
@@ -156,6 +156,14 @@ static int do_mcopy(const char *node, const char *root) {
                 if (!p)
                         return log_oom();
 
+                if (fstatat(dirfd(rootdir), de->d_name, &st, AT_SYMLINK_NOFOLLOW) < 0)
+                        return log_error_errno(errno, "Failed to stat %s: %m", p);
+
+                if (S_ISLNK(st.st_mode)) {
+                        log_debug("%s is a symlink which are not supported by vfat, ignoring", p);
+                        continue;
+                }
```

Comment:
hmm, instead of fstat()ing this always, let's use `de->d_type == DT_LNK` if it is set, and only use fstat() if that is set to DT_UNKNOWN. 

in fact, use dirent_ensure_type() to automatically fill in d_type for you if missing.

Reply:
(this should make a massive speed difference btw, because instead of a likely a single syscall for the entire dir you might end up making up one for each dirent unnecessarily)

Reply:
(to my knowledge all sane file systems initialize d_type btw, but reiserfs does not, and I think vfat neither?)